### ISSUE TITLE
Can multi-layer file nesting be handled?

### ISSUE THREAD

**Issue by UserZhangXiaoZhe** [docwire/docwire#125]
Can multi-layer file nesting be handled? For example, if a Word file or a PPT file contains a PNG image, can the content of the PNG be extracted? Does compressed file include compressed files?

**Comment by as-ascii** [comment]
Yes, it is possible by design. Every parser like DOC parser, DOCX parser, EML parser or others can emit tag::File during the parsing process and it will be processed by other parsers. Recursion will also work. And OCR can also work.

Automatic tests covers such scenarios like:
- attachments in EML file (email)
- emails in PST file (Outlook/Exchange folders)
- XLS embedded inside DOC file (OLE)

Unfortunately not all parsers emit necessary signals currently. We plan to add it but a queue of work with the SDK is very large as you can imagine. It will improve over time.

If you have specific requirements that needs to be covered ASAP please write it down. We are always considering user-voting on specific features and if project is important, the work is large and the term is short our sales team can contact you negotiating some commercial-contract-work.

Best regards.

**Comment by as-ascii** [comment]
Images support in PDF files was added in this release: https://github.com/docwire/docwire/releases/tag/2025.05.22
They can be processed by OCRParser in the same parsing chain.
SDK will be improved in this area in future releases for sure.
